core/src/main/scala/org/apache/spark/sql/delta/commands/ShowTableColumnsCommand.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.commands
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.delta.{DeltaErrors, DeltaLog}
+import org.apache.spark.sql.execution.command.LeafRunnableCommand
+
+/**
+ * The column format of the result returned by the `SHOW COLUMNS` command.
+ */
+case class TableColumns(col_name: String)
+
+/**
+ * A command listing all column names of a table.
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   SHOW COLUMNS (FROM | IN) tableName [(FROM | IN) schemaName];
+ * }}}
+ *
+ * @param path  the file path where the Delta table located
+ */
+case class ShowTableColumnsCommand(path: Path)
+  extends LeafRunnableCommand with DeltaCommand {
+
+  override val output: Seq[Attribute] = ExpressionEncoder[TableColumns]().schema.toAttributes
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    // Return the schema from snapshot if it is an Delta table. Or raise `fileNotFoundException` if
+    // it is a non-Delta table.
+    val deltaLog = DeltaLog.forTable(sparkSession, path)
+    recordDeltaOperation(deltaLog, "delta.ddl.showColumns") {
+      if (deltaLog.snapshot.version < 0) {
+        throw DeltaErrors.fileNotFoundException(path.toString)
+      } else {
+        deltaLog.snapshot.schema.fieldNames.map { x => Row(x) }.toSeq
```

Comment:
Looks like there isn't much code here. Can this be converted into a anonymous object in DeltaSqlParser that just returns the rows?

Reply:
ignore this comment.